lib/deploy.go

```diff
@@ -99,6 +100,28 @@ func deploy(c *cli.Context, name string, gapiObj gapi.GAPI) error {
 		}
 	}
 
+	// be helpful when using unix domain socket urls
+	// unix:// url path is relative (and does not support directories), so mgmt creates the
+	// socket files in the prefix directory, switch to default or user provided prefix directory
+	var unixsockets bool
+	for _, v := range c.GlobalStringSlice("seeds") {
+		if strings.HasPrefix(v, "unix://") {
+			unixsockets = true
+		}
+	}
+	if unixsockets {
+		wd, err := os.Getwd()
+		if err != nil {
+			return errwrap.Wrapf(err, "could not get current working directory")
+		}
+		prefix := c.GlobalString("prefix")
+		log.Printf("Deploy: changing to directory %s", prefix)
+		if err := os.Chdir(prefix); err != nil {
```

Comment:
I'm concerned this might cause a problem for other parts of the code that might be expecting the original working directory. Do you think that's the case? Is there a more isolated way of doing this, or can we do so using absolute paths so we don't need to `os.Chdir` ?

Reply:
There is a defer to change back to the original directory after the enclosing code block is complete. I will check if etcd/grpc has been updated to support non-relative `unix://` paths in the meantime. Then we can drop this entire logic.

Reply:
That would be excellent! I just honestly don't know if it would affect other goroutines or if the chdir affects the entire program. My guess is the latter, which would potentially be problematic for any case that looks at the working dir.

Reply:
The problem still persists in etcd, in that if you specify a `unix://` socket it tries to unpack it into a host and port. So `unix://clients.sock:0` is unpacked to 'host' `clients.sock` and 'port' `0`. It will result in the socket file `clients.sock:0` being created in the current working directory. Adding any path before the file causes an error of it not being a 'host'.

The logic here is only invoked when unix sockets are specified as client or server urls. So unless a user explicitly specifies these in arguments (or runs with `--no-network`) this code won't trigger.

As far as the `Chdir` goes. It is only needed when etcd creates the socket files. After that we can chdir back to the original working directory. I could wrap this logic around the etcd clients creation if you want to make it more explicit. Then the chdir back would happen right after instead of at the end of the function.

Reply:
So it seems the Chdir thing is a problem, see: https://github.com/golang/go/issues/27658
So can we open a bug upstream wherever that socket is written so that it uses an absolute path?

Reply:
That might be a problem indeed. I think it would be a slim chance since this is called early on in the program so not many other goroutines relying on the working dir should be active. But I don't know for certain so there might be the odd race condition.

This part of the code is only used if you want to deploy mcl on the host itself to a non-networking instance of mgmt. So functionally it could be replaced with restarting mgmt instance after you change the configuration files locally.